File: fix_all_imports.py
#!/usr/bin/env python3
"""
Comprehensive import statement fixes for React components
"""

import os
import re
import glob
from pathlib import Path

def fix_all_imports(file_path):
    """Fix all import statement issues in React components"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Remove all malformed import lines
        lines = content.split('\n')
        clean_lines = []
        
        for line in lines:
            # Skip malformed import lines
            if (line.strip().startswith('import ') and 
                ('{ {' in line or '} from \'' in line or '}\'  }' in line or 
                 'react-router-dom\'' in line or 'lucide-react\'' in line or
                 'react-helmet-async\'' in line)):
                continue
            clean_lines.append(line)
        
        content = '\n'.join(clean_lines)
        
        # Add proper imports at the top
        imports = []
        if 'Helmet' in content:
            imports.append("import { Helmet } from 'react-helmet-async';")
        if 'Link' in content:
            imports.append("import { Link } from 'react-router-dom';")
        if 'ArrowRight' in content or 'Home' in content or 'Search' in content or 'AlertTriangle' in content:
            imports.append("import { ArrowRight, Home, Search, AlertTriangle } from 'lucide-react';")
        
        # Remove duplicates
        imports = list(set(imports))
        
        # Add imports to the beginning
        if imports:
            content = '\n'.join(imports) + '\n\n' + content
        
        # Clean up extra whitespace
        content = re.sub(r'\n\s*\n\s*\n+', '\n\n', content)
        
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            return True
                
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False
    
    return False

def main():
    """Main function to fix all imports"""
    workspace = Path('/workspace')
    
    # File extensions to process
    extensions = ['*.tsx', '*.jsx']
    
    total_files = 0
    fixed_files = 0
    
    print("Starting comprehensive import fixes...")
    
    for ext in extensions:
        pattern = workspace / 'app' / '**' / ext
        files = glob.glob(str(pattern), recursive=True)
        
        for file_path in files:
            # Skip specific files
            if any(skip in file_path for skip in ['node_modules', '.git', 'dist', '.next', 'out']):
                continue
                
            total_files += 1
            
            if fix_all_imports(file_path):
                fixed_files += 1
                print(f"Fixed: {file_path}")
    
    print(f"\nComprehensive import fixes complete!")
    print(f"Total files processed: {total_files}")
    print(f"Files fixed: {fixed_files}")

if __name__ == "__main__":
    main()

File: fix_all_imports.js
#!/usr/bin/env node

import fs from 'fs';
import { execSync } from 'child_process';

// Get all files with named ErrorBoundary imports
const getFilesWithNamedImports = () => {
  try {
    const result = execSync('find app -name "*.tsx" -exec grep -l "import { ErrorBoundary }" {} \\;', { encoding: 'utf8' });
    return result.trim().split('\n').filter(file => file.trim());
  } catch (error) {
    return [];
  }
};

// Fix import statements
const fixImports = (filePath) => {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    const originalContent = content;
    
    // Replace named import with default import
    content = content.replace(
      /import { ErrorBoundary } from ['"]\.\.\/components\/ErrorBoundary['"];?/g,
      'import ErrorBoundary from \'../components/ErrorBoundary\';'
    );
    
    // Also handle cases with different path depths
    content = content.replace(
      /import { ErrorBoundary } from ['"]\.\.\/\.\.\/components\/ErrorBoundary['"];?/g,
      'import ErrorBoundary from \'../../components/ErrorBoundary\';'
    );
    
    if (content !== originalContent) {
      fs.writeFileSync(filePath, content, 'utf8');
      return true;
    }
    
    return false;
  } catch (error) {
    console.error(`Error fixing imports in ${filePath}:`, error.message);
    return false;
  }
};

// Main execution
const filesWithNamedImports = getFilesWithNamedImports();
console.log(`Found ${filesWithNamedImports.length} files with named ErrorBoundary imports`);

let fixedCount = 0;
let errorCount = 0;

filesWithNamedImports.forEach(filePath => {
  try {
    if (fixImports(filePath)) {
      fixedCount++;
      console.log(`✓ Fixed imports: ${filePath}`);
    }
  } catch (error) {
    console.error(`✗ Error processing ${filePath}:`, error.message);
    errorCount++;
  }
});

console.log(`\nSummary:`);
console.log(`- Files processed: ${filesWithNamedImports.length}`);
console.log(`- Files fixed: ${fixedCount}`);
console.log(`- Errors: ${errorCount}`);